zerver/lib/push_notifications.py

```diff
@@ -322,14 +322,34 @@ def remove_push_device_token(user_profile, token_str, kind):
 #
 # Push notifications in general
 #
+def get_trigger_from_message(message):
+    # type: (Message) -> Text
+    """
+    Determine what trigger string to send based on missed messages.
+    """
+    if message.recipient.type == Recipient.HUDDLE and message.triggers['private_message'] and message.triggers['mentioned']:
+        return "private_group_mentioned"
+    elif message.recipient.type == Recipient.HUDDLE and message.triggers['private_message']:
+        return "private_group"
+    elif message.recipient.type == Recipient.PERSONAL and message.triggers['private_message']:
+        return "private"
+    elif message.recipient.type == Recipient.STREAM and message.triggers['mentioned']:
+        return "mention_stream"
+    elif (message.recipient.type == Recipient.STREAM and
+            (message.triggers['stream_push_notify'] and message.stream_name)):
+        return "stream"
+    else:
+        return "multiple"
```

Comment:
I actually had been thinking about changing the structure of the original `triggers` data set to just be the single highest priority trigger in the backend, i.e. just passing private, mentioned, or stream_push_notify as the trigger.  What do you think about that? 

I guess my thinking is that it's not actually interesting whether or not a user was mentioned in a group PM, and that would make the protocol a lot simpler.

Reply:
This is the original data you're referring to right https://github.com/zulip/zulip/blob/master/zerver/lib/push_notifications.py#L406 ?

So for this data, yeah I also think this should be simplified to string constants and sent directly.

> I guess my thinking is that it's not actually interesting whether or not a user was mentioned in a group PM, and that would make the protocol a lot simpler.

I think, if we are giving the user information about mentioning on a stream, then we should give the information about a mention on the group, and it's really just adding a trigger constant so it's the work required is also very less!

Reply:
@timabbott What about this one?